lldb/source/ValueObject/DILEval.cpp

```diff
@@ -1852,30 +1995,37 @@ Interpreter::Visit(const UnaryOpNode *node) {
     if (error.Fail())
       return error.ToError();
   }
+  CompilerType rhs_type = rhs->GetCompilerType();
 
   switch (node->kind()) {
-    case UnaryOpKind::Deref:
-      {
-        lldb::ValueObjectSP dynamic_rhs =
-            rhs->GetDynamicValue(m_default_dynamic);
-        if (dynamic_rhs)
-          rhs = dynamic_rhs;
-
-        if (rhs->GetCompilerType().IsPointerType())
-          return EvaluateDereference(rhs);
-        lldb::ValueObjectSP child_sp = rhs->Dereference(error);
-        if (error.Success())
-          rhs = child_sp;
+    case UnaryOpKind::Deref: {
+      if (rhs_type.IsArrayType()) {
+        rhs = ArrayToPointerConversion(rhs, m_exe_ctx_scope);
+      }
```

Comment:
llvm style comment: Don't use curly braces around a single statement.

https://llvm.org/docs/CodingStandards.html#don-t-use-braces-on-simple-single-statement-bodies-of-if-else-loop-statements